docs/guides/custom-functions.md

```diff
@@ -107,6 +114,43 @@ rules:
       function: "abc"
 ```
 
+#### Async Function Example
+
+**functions/dictionary.js**
+
+```js
+const CACHE_KEY = 'dictionary';
+
+module.exports = async function (targetVal) {
+  if (!this.cache.has(CACHE_KEY)) {
+    const res = await fetch('https://dictionary.com/evil');
+    if (res.ok) {
+      this.cache.set(CACHE_KEY, await res.json());
+    } else {
+      // you can either re-try or just throw an error
```

Comment:
@P0lip What's the effect from throwing from within a function? Does the whole process crash down? Do we get a special kind of result or a console log?

From the function writer standpoint, what kind of code should be written 
- if I don't want this failure being unnoticed
- if I prefer to see a CI build fail rather than having it pass for the wrong reasons

_Note:_ I know this is kind off-topic with regards to the scope of the PR but... well... this is documentation and, to be honest, the doco actually lacks guidance about this.

_Note 2:_ If you prefer this to be dealt with later, could you please log an issue to keep track of that?

Reply:
> @P0lip What's the effect from throwing from within a function? Does the whole process crash down? Do we get a special kind of result or a console log?

We'd get a console.log, same as we do when any other custom function throws an exception.
The whole process would still keep on going.

> if I prefer to see a CI build fail rather than having it pass for the wrong reasons

I'd return a regular error, i.e.
```
return [{ message: 'Network error. The check could not be performed' }]
```

Reply:
@P0lip That's very valuable info. I've logged #1096 as I think this deserves a doco section of its own.